### PR TITLE
first commit for part 1

### PR DESCRIPTION
**Pull request by rtaub5**

### CODE REVIEW

File: src/main/java/taub/citi/Main.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package org.example;
+package taub.citi;
```
Comment (andrewoid) [inline src/main/java/taub/citi/Main.java]: Is this file necessary?

File: .idea/gradle.xml
Comment (andrewoid) [inline .idea/gradle.xml:1]: Should not be in git.

Your gitignore should exclude everything in you .idea directory.

File: src/main/java/taub/citi/Station.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
+package taub.citi;
+
+public class Station
+{
+    //CHECKSTYLE:OFF
+    public String station_type;
```
Comment (andrewoid) [inline src/main/java/taub/citi/Station.java:6]: Only include the variables you need to use. Not everything.

File: src/test/java/taub/citi/CitiBikeServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+        Stations stations = service.getStatuses().blockingGet();
+        Station station1 = stations.data.stations.get(0);
+        assertNotNull(station1.station_id);
+        assertNotNull(station1.num_docks_available);
+        assertNotNull(station1.last_reported);
+    }
+
+    @Test
+    void getStatusByStationId()
```
Comment (andrewoid) [inline src/test/java/taub/citi/CitiBikeServiceTest.java:36]: You're not actually testing anything with this.

Try creating a method that does this and then having your test call the method.

File: src/test/java/taub/citi/CitiBikeServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+                if (station.num_docks_available != 0)
+                {
+                    combinedStations.add(station);
+                }
+            }
+        }
+        return combinedStations;
+    }
+
+    public Station findClosestStation(double lat, double lon, List<Station> stations)
```
Comment (andrewoid) [inline src/test/java/taub/citi/CitiBikeServiceTest.java]: This code should not be in your test. You will need to use this later in your lambda.

File: src/test/java/taub/citi/CitiBikeServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+    }
+
+    @Test
+    void getClosestAvailableBikeLocation()
+    {
+        double currLat = 40.5;
+        double currLon = -73.1;
+        Stations stationInfo = service.getStations().blockingGet();
+        Stations stationStatus = service.getStatuses().blockingGet();
+        int retval = 1; // finding the closest bike
```
Comment (andrewoid) [inline src/test/java/taub/citi/CitiBikeServiceTest.java]: Terrible name for a variable since it's not actually a return value

File: .idea/gradle.xml
Comment (rtaub5) [reply .idea/gradle.xml:1]: I'm not sure why it's included. I included the specified lines in my gitignore.
Screenshot 2024-12-30 at 8 29 25 PM

Comment (andrewoid) [reply .idea/gradle.xml:1]: You need to remove it from the git ignore then delete the files.

Then commit

Then as the files back into the git ignore

Comment (rtaub5) [reply .idea/gradle.xml:1]: Remove the lines, and the files. Commit, and put back the files and the lines in the gitignore? Or should I keep the files deleted?

File: src/main/java/taub/citi/Station.java (hunk 2 above)
Comment (rtaub5) [reply src/main/java/taub/citi/Station.java:6]: What if I am not sure what I need yet? Can I keep it in for now?

Comment (andrewoid) [reply src/main/java/taub/citi/Station.java:6]: Ok...

Comment (rtaub5) [reply src/main/java/taub/citi/Station.java:6]: I fixed the gitignore and the other comments. Am I able to merge my pr?

File: src/main/java/taub/citi/Stations.java [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+                currStation = station;
+                break;
+            }
+        }
+        return currStation;
+    }
+
+    // usage is to call the method from the list of stationInfos and pass in the stationstatuses
+    // (This method won't work in the opposite order)
+    public Stations combineStationInfo(List<Station> stationStatus, int retVal)
```
Comment (andrewoid) [inline src/main/java/taub/citi/Stations.java]: Rename retval to something that actually describes what it is

Comment (rtaub5) [reply src/main/java/taub/citi/Stations.java]: I fixed this. Is everything else good?

File: src/main/java/taub/citi/Stations.java [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+                {
+                    combinedStations.add(station);
+                }
+            } else if (bikeOrSlotSelection == 2) // if finding the closest dock, omit stations with no docks available
+            {
+                if (station.num_docks_available != 0)
+                {
+                    combinedStations.add(station);
+                }
+            }
```
Comment (andrewoid) [inline src/main/java/taub/citi/Stations.java]: Make 1 and 2 named constants

Simplify these of statements by combining them

**Review by andrewoid — APPROVED** [review]
